internal/jobcontainers/storage.go

```diff
@@ -0,0 +1,93 @@
+package jobcontainers
+
+import (
+	"context"
+	"os"
+	"path/filepath"
+
+	"github.com/Microsoft/hcsshim/internal/layers"
+	"github.com/Microsoft/hcsshim/internal/log"
+	"github.com/Microsoft/hcsshim/internal/wclayer"
+	specs "github.com/opencontainers/runtime-spec/specs-go"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/sys/windows"
+)
+
+// Trailing backslash required for SetVolumeMountPoint and DeleteVolumeMountPoint
+const sandboxMountFormat = `C:\C\%s\`
+
+func mountLayers(ctx context.Context, s *specs.Spec) error {
+	if s == nil || s.Windows == nil || s.Windows.LayerFolders == nil {
+		return errors.New("field 'Spec.Windows.Layerfolders' is not populated")
+	}
+
+	// Last layer always contains the sandbox.vhdx, or 'scratch' space for the container.
+	scratchFolder := s.Windows.LayerFolders[len(s.Windows.LayerFolders)-1]
+	if _, err := os.Stat(scratchFolder); os.IsNotExist(err) {
+		if err := os.MkdirAll(scratchFolder, 0777); err != nil {
+			return errors.Wrapf(err, "failed to auto-create container scratch folder %s", scratchFolder)
+		}
+	}
+
+	// Create sandbox.vhdx if it doesn't exist in the scratch folder.
+	if _, err := os.Stat(filepath.Join(scratchFolder, "sandbox.vhdx")); os.IsNotExist(err) {
+		if err := wclayer.CreateScratchLayer(ctx, scratchFolder, s.Windows.LayerFolders[:len(s.Windows.LayerFolders)-1]); err != nil {
+			return errors.Wrap(err, "failed to CreateSandboxLayer")
+		}
+	}
```

Comment:
With your changes to WCOW scratch in containerd we will always create a scratch vhd for a snapshot  right? If that's the case then do we need to keep this code here?

Reply:
No, but that change still isn't in yet so I've left it for now. I can just quickly remove it whenever depending on what gets in first